core/utils/mailbox_prom.go

```diff
@@ -0,0 +1,49 @@
+package utils
+
+import (
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var mailboxLoad = promauto.NewGaugeVec(prometheus.GaugeOpts{
+	Name: "mailbox_load_percent",
+	Help: "Percent of mailbox capacity used",
+},
+	[]string{"name", "capacity"},
+)
+
+const mailboxPromInterval = 10 * time.Second
+
+var (
+	mailboxes   sync.Map
+	monitorOnce sync.Once
+)
+
+type loadFn func() (capacity uint64, percent float64)
+
+func monitor(name string, load loadFn) {
+	mailboxes.Store(name, load)
+	monitorOnce.Do(startMonitoring)
+}
+
+func unMonitor(name string) {
+	mailboxes.Delete(name)
```

Comment:
Perhaps we want to stop monitoring if its empty? Thinking about dangling goroutines after tests and if that might cause problems

Reply:
I considered that, but we'd have to give up sync.Map and use a coarse lock, and it could thrash on and off. Leaving one single nop goroutine seemed low impact, but I'm still fiddling with it :shrug: 
Another earlier version let callers pass a `stop <-chan` to `monitor`, but it felt odd because the first call wins and it could be accidentally silently shut off and never turn back on again.